nni/retiarii/execution/api.py

```diff
@@ -21,7 +23,10 @@ def get_execution_engine() -> BaseExecutionEngine:
     """
     global _execution_engine
     if _execution_engine is None:
-        _execution_engine = BaseExecutionEngine()
+        if os.environ.get('CGO') == 'true':
```

Comment:
another option is let strategy pass engine type as an input argument to this API